test/DynamicHelpTest.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Management.Automation;
+using Xunit;
+
+namespace Test
+{
+    public partial class ReadLine
+    {
+        private static readonly string fullHelp = @"
+
+NAME
+    Get-Date
+
+SYNOPSIS
+    Gets the current date and time.
+
+PARAMETERS
+    -AsUTC <System.Management.Automation.SwitchParameter>
+        Converts the date value to the equivalent time in UTC.
+
+        This parameter was introduced in PowerShell 7.1.
+
+        Required?                    false
+        Position?                    named
+        Default value                False
+        Accept pipeline input?       False
+        Accept wildcard characters?  false
+";
+
+        internal static object GetDynamicHelpContent(string commandName, string parameterName, bool isFullHelp)
+        {
+            if (string.IsNullOrEmpty(commandName))
+            {
+                return null;
+            }
+
+            if (isFullHelp)
+            {
+                return fullHelp;
+            }
+
+            if (string.IsNullOrEmpty(parameterName))
+            {
+                return null;
+            }
+
+            PSObject paramHelp = new PSObject();
+
+            if (string.Equals(commandName, "Get-FakeHelp", StringComparison.OrdinalIgnoreCase) && string.Equals(parameterName, "Fake", StringComparison.OrdinalIgnoreCase))
+            {
+                PSObject[] descDetails = new PSObject[1];
+                descDetails[0] = new PSObject();
+                descDetails[0].Members.Add(new PSNoteProperty("Text", null));
+            }
+            else if (string.Equals(commandName, "Get-Date", StringComparison.OrdinalIgnoreCase) && string.Equals(parameterName, "Date", StringComparison.OrdinalIgnoreCase))
+            {
+                return GetParameterHelpObject(description: "Specifies a date and time.");
+            }
+            else if (string.Equals(commandName, "Get-MultiLineHelp", StringComparison.OrdinalIgnoreCase))
+            {
+                if(string.Equals(parameterName, "OneAndHalf", StringComparison.OrdinalIgnoreCase))
+                {
+                    string multiLineDesc = "Some very long description that is over the buffer width of 60 characters but shorter than 120.";
+                    return GetParameterHelpObject(multiLineDesc);
+                }
+                else if(string.Equals(parameterName, "ExactlyTwo", StringComparison.OrdinalIgnoreCase))
```

Comment:
There is no test to verify the `-ExactlyTwo` parameter help.

Reply:
Also, the key bindings are enabled for Emacs and VI modes too, so maybe add a few more tests to verify them in those edit modes.